### ISSUE TITLE
`nim --version` doesn't show git hash (when installed via `choosenim '#head'`)

### ISSUE THREAD

**Issue by timotheecour** [dom96/choosenim#57]
```
./bin/choosenim --nimbleDir:$HOME/.nimble_choosenim '#head'
   Switched to Nim #head

 ~/.nimble_choosenim/bin/nim --version
Nim Compiler Version 0.18.1 [MacOSX: amd64]
Copyright (c) 2006-2018 by Andreas Rumpf

active boot switches: -d:release
```

but

```
brew install --HEAD nim
nim --version
Nim Compiler Version 0.18.1 [MacOSX: amd64]
Copyright (c) 2006-2018 by Andreas Rumpf

git hash: 992300b30057e2b3b489b90fb0e7011607e3e8cf
active boot switches: -d:release -d:useLinenoise
```

=> ` ~/.nimble_choosenim/bin/nim --version` should show git hash, as is the case for nim installed by brew

**Comment by dom96** [comment]
choosenim doesn't use git to download Nim (for now). And there isn't anything that can be done to fix this for stable versions.

**Comment by timotheecour** [comment]
shouldn't this work?
```
  # add this to what `--version` prints:
  const gitHash = staticExec "git rev-parse HEAD"
```
also, with a `-dirty` suffix in case `git status` is not clean

**Comment by timotheecour** [comment]
@dom96

> choosenim doesn't use git to download Nim (for now). And there isn't anything that can be done to fix this for stable versions.

actually there is: in download.nim, instead of downloading 'https://github.com/nim-lang/Nim/archive/branch.zip' , download  'https://github.com/nim-lang/Nim/archive/branch~0.zip'
and then look at the returned messages, and parse the hash out of it.

eg on cmd line:
```
wget 'https://github.com/nim-lang/Nim/archive/devel~0.zip'
--2018-04-11 13:59:36--  https://github.com/nim-lang/Nim/archive/devel~0.zip
Resolving github.com (github.com)... 192.30.255.112, 192.30.255.113
Connecting to github.com (github.com)|192.30.255.112|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://codeload.github.com/nim-lang/Nim/zip/e7edc7ec7f5990c80f0bb1ff9d1cd7f0b000bb0e [following]
--2018-04-11 13:59:36--  https://codeload.github.com/nim-lang/Nim/zip/e7edc7ec7f5990c80f0bb1ff9d1cd7f0b000bb0e
Resolving codeload.github.com (codeload.github.com)... 192.30.255.120, 192.30.255.121
Connecting to codeload.github.com (codeload.github.com)|192.30.255.120|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 20421875 (19M) [application/zip]
Saving to: ‘devel~0.zip’

devel~0.zip                                               100%[=====================================================================================================================================>]  19.48M  17.4MB/s    in 1.1s

2018-04-11 13:59:37 (17.4 MB/s) - ‘devel~0.zip’ saved [20421875/20421875]
```

=> git hash is `e7edc7ec7f5990c80f0bb1ff9d1cd7f0b000bb0e`

- [ ] NOTE
same trick can be used for other downloads, eg github download understands these:
devel~5 or master^^ etc

- [ ] BUG
```
git log -n 15 --format='%H'|cat
e7edc7ec7f5990c80f0bb1ff9d1cd7f0b000bb0e
6baca58693672f8fc9485272b3c350d0b0b77163
1ee5f1e36afd31f733195b13e638354ec9412948
4887f641077c93295ffa2a0d7482a73a87ed67d4
f6c8f97fe8bd22e7a2c1b9463f52747e98693682
16c1a90857992df0960f53f5ffa578f14059cee4
427490a845afa13c460d71c506994a24aae900c8
992300b30057e2b3b489b90fb0e7011607e3e8cf
c6e08725c1be978d644547f5140ab3301538236e
56dc7c6f777b0dbd2fd65d04e39f0e7e34b0ca32
673f5f3534cf0fd4e1316d1b7fede31f10878b46
8ae0ac8637dcb11fe60f365651ecfe8d20e8ced2
d6793ded27537b51bbcc1dbdfacaf6f7655bf289
8b7c2bd06783c916c9f3725ee571bf2463140a73
2d686743386076a3edbfd1db4723cf9f3064f08c
```

wget 'https://github.com/nim-lang/Nim/archive/devel~8.zip'
predictably shows the 8th entry (starting from index 0)`c6e08725c1be978d644547f5140ab3301538236e`
however, wget 'https://github.com/nim-lang/Nim/archive/devel~9.zip' shows the 10th entry(starting from index 0) `673f5f3534cf0fd4e1316d1b7fede31f10878b46` 
not sure why

**Comment by dom96** [comment]
> /cc @dom96 do you mind re-opening this?

I do. Once #12 is implemented you will get your git hash. I'm not going to spend time working around this problem when a better solution is to spend time implementing #12.

> why not use git clone to some location instead of download (eg as done by homebrew)? that would minimize traffic and simplify the whole logic. However even with download we can do better:

This is what #12 is about.

> actually there is: in download.nim, instead of downloading 'https://github.com/nim-lang/Nim/archive/branch.zip' , download 'https://github.com/nim-lang/Nim/archive/branch~0.zip'
and then look at the returned messages, and parse the hash out of it.

You want choosenim to patch ``nim`` with the git hash? 

FYI if you download and install https://nim-lang.org/download/nim-0.18.0.tar.xz you also won't have a git hash in there, because it's a stable release.

**Comment by genotrance** [comment]
This is fixed in #173 for devel / #head.
